File: javascript/isbn-verifier/isbn-verifier.js
export const isValid = (input) => {

    let isbn = input.replaceAll(/[^X\d]/g, '');

    if (!/^\d{9}[\dX]$/.test(isbn)) {
        return false;
    }

    return [...isbn].reduce((acc, number, index) => 
        ((number === 'X') ? 10 : Number(number)) * (10 - index) + acc, 0) 
            % 11 === 0;
};

File: python/isbn-verifier/isbn_verifier.py
import string

def is_valid(isbn):

    isbn = isbn.replace("-", "")

    if len(isbn) != 10:
        return False

    numbers = list(isbn)

    if numbers[-1] == "X":
        numbers[-1] = "10"

    res = 0

    for index, number in enumerate(reversed(numbers), 1):
        if number in string.ascii_letters:
            return False
        res += int(number) * index

    return res % 11 == 0
